### ISSUE TITLE
Accusative Direction Semantics: Learn motion vs location

### ISSUE THREAD

**Issue by marctjones** [marctjones/klareco#105]
## Overview

When accusative case (-n) combines with prepositions, it marks **direction/motion** rather than just object role. This distinction has significant semantic meaning.

## Current State

**In AST**: Case is captured (`kazo: akuzativo`), but direction meaning is not extracted.

```python
# Both parse similarly, but mean different things:
"en la domo" → preposition 'en' + noun in nominative
"en la domon" → preposition 'en' + noun in accusative
```

### Location vs Motion
- "Li estas en la domo" (He is in the house) - static location
- "Li iras en la domon" (He goes into the house) - motion toward

### Same Preposition, Different Meaning
| Preposition | + Nominative (Location) | + Accusative (Direction) |
|-------------|------------------------|--------------------------|
| en | in, inside | into |
| sur | on | onto |
| sub | under | to under |
| ĉirkaŭ | around (static) | around (motion) |
| tra | through (static) | through (motion) |

### Semantic Distinction
- "La kato saltas sur la tablon" = cat jumps ONTO the table
- "La kato saltas sur la tablo" = cat jumps WHILE ON the table

These have very different meanings\!

### 1. Detection (Deterministic)
```python
def detect_direction(ast):
    """Check if accusative marks direction with preposition."""
    for i, word in enumerate(ast['aliaj']):
        if word.get('vortspeco') == 'prepozicio':
            # Look for following accusative noun
            for j in range(i+1, len(ast['aliaj'])):
                next_word = ast['aliaj'][j]
                if next_word.get('kazo') == 'akuzativo':
                    return True, word['radiko']  # (has_direction, preposition)
    return False, None
```

Add to AST:
```python
ast['direkto'] = {
    'havas': True,
    'prepozicio': 'en',
    'celo': 'domo'  # target of motion
}
```

### 2. Semantic Effect (Learned)
```python
class DirectionTransformer(nn.Module):
    def __init__(self, dim):
        self.location_to_direction = nn.Linear(dim, dim)
    
    def forward(self, prep_embedding, noun_embedding, is_direction):
        combined = prep_embedding + noun_embedding
        if is_direction:
            return self.location_to_direction(combined)
        return combined
```

### 3. Training Pairs
```python
# Location vs direction - different meanings
("en la domo", "en la domon", similarity=0.4)

# Same preposition, same case - should be similar
("en la domon", "sur la tablon", similarity=0.6)  # Both motion

# Motion verbs should cluster with direction phrases
("Li iras en la domon", "Li kuras al la domo", similarity=0.7)
```

### 4. Evaluation Metrics
| Metric | Target |
|--------|--------|
| Direction detection accuracy | >95% |
| Location vs motion discrimination | >80% |
| Motion phrase clustering | Consistent |

## Prepositions That Take Direction

All prepositions can take accusative for direction, but most common:
- en (in/into)
- sur (on/onto)  
- sub (under/to under)
- al (to/toward) - always direction, no location sense
- ĝis (until/up to)
- tra (through)
- trans (across)
- ĉirkaŭ (around)
- inter (between/among)

## Related Issues

- #101 Meta: Grammatical Features with Semantic Content (parent)
- #89 Parser: preposition 'por' not recognized (parser prerequisite)
